plugin/src/main/scala/eu/unicredit/swagger/SwaggerCodegenPlugin.scala

```diff
@@ -72,166 +76,100 @@ object SwaggerCodegenPlugin extends AutoPlugin {
     /*
      * Tasks
      */
-    val swaggerCleanTask = taskKey[Unit]("Clean swagger generated packages")
+    val swaggerClean = taskKey[Unit]("Clean swagger generated packages")
 
-    val swaggerCodeGenTask = taskKey[Unit]("Generate swagger models and json converters")
+    val swaggerCodeGen = taskKey[Unit]("Generate swagger models and json converters")
 
-    val swaggerServerCodeGenTask = taskKey[Unit]("Generate swagger server routes and controllers boilerplate")
+    val swaggerServerCodeGen = taskKey[Unit]("Generate swagger server routes and controllers boilerplate")
 
-    val swaggerClientCodeGenTask = taskKey[Unit]("Generate swagger client class with WS calls to specific routes")
+    val swaggerClientCodeGen = taskKey[Unit]("Generate swagger client class with WS calls to specific routes")
 
   }
 
-  final val swaggerCodeGenPackageDefault = "swagger.codegen"
-  final val swaggerSourcesDirDefault = s"${separator}src${separator}main${separator}swagger"
-  final val swaggerModelCodeTargetDirDefault = s"${separator}src${separator}main${separator}scala"
-  final val swaggerClientCodeTargetDirDefault = s"${separator}src${separator}main${separator}scala"
-  final val swaggerServerRoutesFileDefault = s"${separator}src${separator}main${separator}resources${separator}routes"
-  final val swaggerModelFilesSplittingDefault = "singleFile"
-  final val swaggerGenerateJsonRWDefault = true
-  final val swaggerGenerateControllersDefault = true
-  final val swaggerCodeProvidedPackageDefault = "com.yourcompany"
-
-  final val swaggerModelCodeGenClassDefault = new DefaultModelGenerator()
-  final val swaggerJsonCodeGenClassDefault = new DefaultJsonGenerator()
-  final val swaggerServerCodeGenClassDefault = new DefaultServerGenerator()
-  final val swaggerClientCodeGenClassDefault = new DefaultClientGenerator()
-
   import autoImport._
   override def trigger = allRequirements
-  override lazy val buildSettings = Seq(
-    commands ++= Seq(
-      swaggerCleanCommand,
-      swaggerCodeGenCommand,
-      swaggerServerCodeGenCommand,
-      swaggerClientCodeGenCommand))
 
   override val projectSettings = {
     Seq(
-    swaggerCleanTask := {
-      val base = baseDirectory.value.getAbsolutePath
-      val codegenPackage = swaggerCodeGenPackage.?.value getOrElse swaggerCodeGenPackageDefault
-
-      swaggerCleanImpl(base, codegenPackage)
+    swaggerSourcesDir           := baseDirectory.value / "src" / "main" / "swagger",
```

Comment:
what would be the advantage of using globalSettings?

for example in a multi-project build, i'd prefer the keys to be known in the one subproject where i actually have swagger bindings to create, but nowhere else.